File: backend/src/main/java/it/unical/inf/ea/backend/dto/OrderDTO.java
package it.unical.inf.ea.backend.dto;


import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@ToString
@Builder
@AllArgsConstructor
public class OrderDTO {

    @NotNull
    private UUID id;

    @NotNull
    private List<OrderItemDTO> items;

    @NotNull
    private BigDecimal totalCost;

    @NotNull
    private String status;

    @NotNull
    private String createdAt;

    @NotNull
    private String updatedAt;

    @NotNull
    private UserDTO user;

    @NotNull
    private AddressDTO address;

    @NotNull
    private PaymentMethodDTO paymentMethod;
}

File: frontend/app/src/main/java/com/android/frontend/dto/OrderDTO.kt
package com.android.frontend.dto

import java.math.BigDecimal
import java.time.LocalDateTime
import java.util.UUID

data class OrderDTO(
    val id: UUID,
    val items: List<OrderItemDTO>,
    val totalCost: BigDecimal,
    val status: String,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
    val user: UserDTO,
    val address: AddressDTO,
    val paymentMethod: PaymentMethodDTO
){
    enum class OrderStatus {
        CREATED,
        PROCESSING,
        SHIPPED,
        DELIVERED,
        CANCELLED,
        RETURNED
    }
}